src/database.c

```diff
@@ -330,18 +330,18 @@ _add_to_db(ProfMessage* message, char* type, const Jid* const from_jid, const Ji
     char* escaped_message = str_replace(message->plain, "'", "''");
 
     if (asprintf(&query, "INSERT INTO `ChatLogs` (`from_jid`, `from_resource`, `to_jid`, `to_resource`, `message`, `timestamp`, `stanza_id`, `archive_id`, `replace_id`, `type`, `encryption`) SELECT '%s', '%s', '%s', '%s', '%s', '%s', '%s', '%s', '%s', '%s', '%s' WHERE NOT EXISTS (SELECT 1 FROM `ChatLogs` WHERE `archive_id` = '%s')",
-                 from_jid->barejid,
+                 from_jid->barejid ? from_jid->barejid : "",
```

Comment:
It's not correct to use this in some cases.
As you can see in https://github.com/profanity-im/profanity/blob/master/src/database.c#L118 some fields are expected to never be NULL. If one of those values is NULL we want sqlitedb to throw an error when we try to insert such a query. I think for this reason "" is not good but needs actually be NULL in those cases. Please change your code accordingly.

Reply:
Also once you do these changes please also change the commit log. The subject line should not exceed 80 chars if possible. In the body you can then be more verbose. There I would also mention the actual file path of the syslog file and what is written in there (just 1 or 2 example lines) before this change.